thirdeye/thirdeye-frontend/app/pods/components/filter-bar/component.js

```diff
@@ -86,17 +87,24 @@ export default Ember.Component.extend({
      * @method filterByEvent
      * @param {Object} clickedBlock - selected filter block object
      */
-    filterByEvent(clickedBlock) {
+    selectEventType(clickedBlock) {
       // Hide all other blocks when one is clicked
-      let filterBlocks = this.get('filterBlocks');
+      let filterBlocks = this.get('config');
+
       filterBlocks.forEach(block => {
         Ember.set(block, 'isHidden', true);
       });
 
       // Note: toggleProperty will not be able to find 'filterBlocks', as it is not an observed property
       // Show clickedBlock
       Ember.set(clickedBlock, 'isHidden', !clickedBlock.isHidden);
-      this.attrs.onTabChange(clickedBlock.eventType);
+
+      const { entities, onSelect } = this.getProperties('entities', 'onSelect');
+      if (onSelect != null) {
```

Comment:
nit: `!==` unless you're checking for undefined and false as well, in which case `if (onSelect)` is sufficient

Reply:
Fixed!